mdoc/src/main/scala-3/mdoc/internal/markdown/CodePrinter.scala

```diff
@@ -0,0 +1,47 @@
+package mdoc.internal.markdown
+
+import java.io.PrintStream
+
+class CodePrinter(ps: PrintStream, baseIndent: Int = 0, baseNest: Int = 0) {
+  private def indent = "  " * (baseIndent + nestCount)
```

Comment:
What happens if the user has 3-space indent?

Reply:
Pushed the test in https://github.com/scalameta/mdoc/pull/466/commits/21980022171cf20f16ca8bf106fe8c24c4605758

Reply:
Looks good. Always worth having tests like these